### PR TITLE
XMSS implementation

### PR DESCRIPTION
**Pull request by SparkiDev**
# Description

Supporting code for wolfSSL C implementation of XMSS.

# Testing

Tests added.

# Checklist

 - [x] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by SparkiDev** [comment]
https://github.com/wolfSSL/pqcrypto/pull/5

**Comment by SparkiDev** [comment]
retest this please

**Review by JacobBarthelmeh — DISMISSED** [review]

File: wolfcrypt/src/error.c [hunk 1]
```diff
@@ -601,6 +601,9 @@ const char* wc_GetErrorString(int error)
     case AES_EAX_AUTH_E:
         return "AES-EAX Authentication check fail";
 
+    case KEY_EXHAUSETED_E:
```
Comment (philljj) [inline wolfcrypt/src/error.c:604]: I like this idea for an error code. Typo though:
`KEY_EXHAUSETED_E` should be `KEY_EXHAUSTED_E`

File: wolfssl/wolfcrypt/error-crypt.h [hunk 2]
```diff
@@ -65,11 +65,17 @@ enum {
     MP_ZERO_E          = -121,  /* got a mp zero result, not expected */
 
     AES_EAX_AUTH_E     = -122, /* AES-EAX Authentication check failure */
+    KEY_EXHAUSETED_E   = -123, /* No longer usable for operation. */
```
Comment (philljj) [inline wolfssl/wolfcrypt/error-crypt.h:68]: Same typo.

Also, is this error code intended for stateful HBS methods only, or just in general for a key with a finite signature space?

File: wolfcrypt/src/error.c (hunk 1 above)
Comment (SparkiDev) [reply wolfcrypt/src/error.c]: Fixed

File: wolfssl/wolfcrypt/error-crypt.h (hunk 2 above)
Comment (SparkiDev) [reply wolfssl/wolfcrypt/error-crypt.h]: Fixed

**Comment by philljj** [comment]
Left review comments offline.

**Comment by SparkiDev** [comment]
Fixed defines for HAVE_LIBXMSS.

**Comment by philljj** [comment]
Retest this please

**Review by philljj — APPROVED** [review]

**Review by JacobBarthelmeh — APPROVED** [review]

**Pull request merged**